src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -82,3 +82,39 @@
            :on-error   #(log/info "failed to remove keypair " %)}]]]})
 
 (rf/reg-event-fx :wallet/remove-keypair remove-keypair)
+
+(defn wallet-validate-seed-phrase
+  [_ [seed-phrase on-success on-error]]
+  {:fx [[:multiaccount/validate-mnemonic [seed-phrase on-success on-error]]]})
+
+(rf/reg-event-fx :wallet/validate-seed-phrase wallet-validate-seed-phrase)
+
+(defn make-seed-phrase-keypair-fully-operable
+  [_ [mnemonic password on-success on-error]]
+  {:fx [[:json-rpc/call
+         [{:method     "accounts_makeSeedPhraseKeypairFullyOperable"
+           :params     [(-> mnemonic security/safe-unmask-data)
+                        (-> password security/safe-unmask-data native-module/sha3)]
+           :on-success [:wallet/make-seed-phrase-keypair-fully-operable-success on-success]
+           :on-error   [:wallet/make-seed-phrase-keypair-fully-operable-error on-error]}]]]})
+
+(rf/reg-event-fx :wallet/make-seed-phrase-keypair-fully-operable make-seed-phrase-keypair-fully-operable)
+
+(defn make-seed-phrase-keypair-fully-operable-success
+  [_ [on-success _result _request-id]]
+  {:fx [[:dispatch [:wallet/get-accounts]]
+        [:dispatch [:wallet/get-keypairs]]
+        [:effects.common/apply-fn [on-success]]]})
+
+(rf/reg-event-fx :wallet/make-seed-phrase-keypair-fully-operable-success
+ make-seed-phrase-keypair-fully-operable-success)
+
+(defn make-seed-phrase-keypair-fully-operable-error
+  [_ [on-error error]]
+  (log/info "failed to make key pair fully operable with seed phrase"
+            {:error error
+             :event :wallet/make-seed-phrase-keypair-fully-operable})
+  {:fx [[:effects.common/apply-fn [on-error error]]]})
+
+(rf/reg-event-fx :wallet/make-seed-phrase-keypair-fully-operable-error
+ make-seed-phrase-keypair-fully-operable-error)
```

Comment:
Here we're defining the events for the `make-seed-phrase-keypair-fully-operable` events.

Reply:
Note that we're using a new effect named `effects.common/apply-fn`. This is because `on-success` and `on-error` are being passed as functions. Perhaps it would be better to send a vector for a re-frame event, though passing functions seems to fit well when combining the components custom behaviour to this generic event.